packages/account-tree-controller/src/wallet.ts

```diff
@@ -15,8 +15,26 @@ import type {
   AccountGroupSingleAccountObject,
 } from './group';
 import { AccountTreeGroup } from './group';
+import type {
+  UpdatableField,
+  ExtractRequiredFieldValues,
+} from './type-utils.js';
 import { type AccountTreeControllerMessenger } from './types';
 
+/**
+ * Persisted metadata for account wallets (stored in controller state for persistence/sync).
+ */
+export type AccountWalletPersistedMetadata = {
+  /** Custom name set by user, overrides default naming logic */
+  name?: UpdatableField<string>;
+};
+
+/**
+ * Tree metadata for account wallets (required plain values extracted from persisted metadata).
+ */
+export type AccountTreeWalletMetadata =
+  ExtractRequiredFieldValues<AccountWalletPersistedMetadata>;
```

Comment:
And same here, we can use `Required` here too:

```suggestion
  Required<ExtractFieldValues<AccountTreeWalletPersistedMetadata>?;
```